WordPress/Classes/ViewRelated/Stats/Insights/Insights Management/InsightsManagementViewController.swift

```diff
@@ -305,11 +312,67 @@ private extension InsightsManagementViewController {
     func toggleRow(for statSection: StatSection) {
         if let index = insightsShown.firstIndex(of: statSection) {
             insightsShown.remove(at: index)
-        } else {
+            moveRowToInactive(at: index, statSection: statSection)
+        } else if let inactiveIndex = insightsInactive.firstIndex(of: statSection) {
             insightsShown.append(statSection)
+            moveRowToActive(at: inactiveIndex, statSection: statSection)
         }
+    }
 
-        reloadViewModel()
+    // Animates the movement of a row from the inactive to active section, supports accessibility
+    func moveRowToActive(at index: Int, statSection: StatSection) {
+        tableHandler.viewModel = tableViewModel()
+
+        let origin = IndexPath(row: index, section: 1)
+        let row = insightsShown.firstIndex(of: statSection) ?? (insightsShown.count - 1)
```

Comment:
nitpick: Since the row was just added to the data source, it seems like `?? (insightsShown.count - 1)` will never be called. Can you explain why this is needed?

Reply:
Thanks for the review, @guarani!

Good observation, it was to avoid having an `optional` and moving the cell to the bottom just-in-case. Although this just-in-case scenario should **never** happen. 

We could use `guard let` and add logging + `tableView.reloadData` to indicate that scenario shouldn't be happening.

Reply:
Gotcha! I don't have any strong feelings about this, happy for you to do what you think is best (even if that's to leave it as-is).

Reply:
I think it's reasonable to leave it as it is. Additional `guard` won't improve the solution much.